flow/core/vehicles.py

```diff
@@ -971,44 +926,83 @@ def set_lane_headways(self, veh_id, lane_headways):
 
     def get_lane_headways(self, veh_id, error=list()):
         """Return the lane headways of the specified vehicles.
-
         This includes the headways between the specified vehicle and the
         vehicle immediately ahead of it in all lanes.
-
         Parameters
         ----------
         veh_id : str or list<str>
             vehicle id, or list of vehicle ids
         error : any, optional
             value that is returned if the vehicle is not found
-
         Returns
         -------
         list<float>
-
         """
         if isinstance(veh_id, (list, np.ndarray)):
             return [self.get_lane_headways(vehID, error) for vehID in veh_id]
         return self.__vehicles.get(veh_id, {}).get("lane_headways", error)
 
+    def get_lane_leaders_speed(self, veh_id, error=list()):
+        """Return the speed of the leaders of the specified vehicles.
+        This includes the speed between the specified vehicle and the
+        vehicle immediately ahead of it in all lanes.
+        Parameters
+        ----------
+        veh_id : str or list<str>
+            vehicle id, or list of vehicle ids
+        error : any, optional
+            value that is returned if the vehicle is not found
+        Returns
+        -------
+        list<float>
+        """
+        if isinstance(veh_id, (list, np.ndarray)):
+            return [self.get_lane_leaders_speed(vehID, error)
+                    for vehID in veh_id]
+        return self.__vehicles.get(veh_id, {}).get("leaders_speed", error)
+
+    def set_lane_leaders_speed(self, veh_id, lead_speed):
+        """Set the speed of the leaders of the specified vehicles"""
+        self.__vehicles[veh_id]["leaders_speed"] = lead_speed
```

Comment:
i don't think there's any point of storing these things like this. u can simply call `self.get_speed(self.get_lane_leaders(veh_id))` in this method